src/NosCore.GameObject/Providers/QuestProvider/QuestProvider.cs

```diff
@@ -219,15 +261,18 @@ public async Task<bool> AddQuestAsync(ICharacterEntity character, short questId)
                 QuestId = quest.QuestId
             });
             await character.SendPacketAsync(character.GenerateQuestPacket()).ConfigureAwait(false);
-
+            if (quest.TargetMap == character.MapId)
+            {
+                await character.SendPacketAsync(quest.GenerateTargetPacket()).ConfigureAwait(false);
+            }
             return true;
         }
 
-        public bool ValidateQuest(ICharacterEntity character, short questId)
+        public async Task<bool> ValidateQuestAsync(ICharacterEntity character, short questId)
         {
             bool isValid = false;
-            var characterQuest = character.Quests.Values.First(s => s.QuestId == questId);
-            switch (characterQuest.Quest.QuestType)
+            var characterQuest = character.Quests.Values.FirstOrDefault(s => s.QuestId == questId);
+            switch (characterQuest?.Quest.QuestType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5304413)